### PR TITLE
Stop using internals: `Base.copymutable` [NFC]

### PR DESCRIPTION
**Pull request by LilithHafner**
I inlined the implementations to the call sites which, remarkably, improves concision.

### CODE REVIEW

File: src/sort.jl [hunk 1]
```diff
@@ -38,12 +38,12 @@ end
 
 
 @inline _sort(a::StaticVector, alg, order) =
-    similar_type(a)(sort!(Base.copymutable(a); alg=alg, order=order))
+    similar_type(a)(sort!(copyo!(similar(a), a); alg=alg, order=order))
```
Comment (thchr) [inline src/sort.jl:41]:
```suggestion
    similar_type(a)(sort!(copyto!(similar(a), a); alg=alg, order=order))
```

Comment (LilithHafner) [reply src/sort.jl]: Haha, thanks!

**Comment by mateuszbaran** [comment]
Why did you replace `Base.copymutable(a)` with `copyto!(similar(a), a)` in one place but `collect(a)` in the other?

**Comment by LilithHafner** [comment]
Because
```julia
julia> @less Base.copymutable([1,2,3])
function copymutable(a::AbstractArray)
    @_propagate_inbounds_meta
    copyto!(similar(a), a)
end
copymutable(itr) = collect(itr)

**Review by mateuszbaran — APPROVED** [review]

**Review by thchr — APPROVED** [review]

**Pull request merged**